src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -400,6 +402,28 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        private async Task<ConnectionCompleteParams> TryOpenConnectionWithRetry(ConnectionInfo connectionInfo, ConnectParams connectionParams)
+        {
+            Boolean isRetriable = true;
+            int counter = 0;
+            ConnectionCompleteParams response = null;
+            while (isRetriable && counter <= MaxTries)

```

Comment:
Make sure there's appropriate logging for each retry. Include the number retry it's on, and have a final message if it gets to the max retries and still isn't able to connect.